### PR TITLE
Spark: Add jobType facet to Spark application events

### PR DESCRIPTION
**Pull request by dolfinus**
### Problem

Currently runEvents for SparkSQL and RDD events have jobType facet, but Spark application events does not:
```json
{
  "eventTime": "2024-05-23T08:03:11.993Z",
  "producer": "https://github.com/OpenLineage/OpenLineage/tree/1.15.0-SNAPSHOT/integration/spark",
  "schemaURL": "https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/RunEvent",
  "eventType": "START",
  "run": {
    "runId": "018fa479-c15d-74fd-a3b6-bf7a05eb2b01"
  },
  "job": {
    "namespace": "spark_integration",
    "name": "onetl"
  },
  "inputs": [],
  "outputs": []
}
```

So it's hard to distinguish events related to Spark application from other generic runEvents.

### Solution

Add `jobType` facet to runEvents emitted by SparkListenerApplicationStart:
```json
{
  "eventTime": "2024-05-23T08:03:11.993Z",
  "producer": "https://github.com/OpenLineage/OpenLineage/tree/1.15.0-SNAPSHOT/integration/spark",
  "schemaURL": "https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/RunEvent",
  "eventType": "START",
  "run": {
    "runId": "018fa479-c15d-74fd-a3b6-bf7a05eb2b01"
  },
  "job": {
    "namespace": "spark_integration",
    "name": "onetl",
    "facets": {
      "jobType": {
        "_producer": "https://github.com/OpenLineage/OpenLineage/tree/1.15.0-SNAPSHOT/integration/spark",
        "_schemaURL": "https://openlineage.io/spec/facets/2-0-2/JobTypeJobFacet.json#/$defs/JobTypeJobFacet",
        "processingType": "BATCH",
        "integration": "SPARK",
        "jobType": "APPLICATION"
      }
    }
  },
  "inputs": [],
  "outputs": []
}
```

#### One-line summary:

Add `jobType` job facet to Spark application events.

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [ ] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [X] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

**Comment by dolfinus** [comment]
Regarding `processingType=BATCH`, Spark application can have both batch and streaming jobs at the same time, so this is arguable. Probably `processingType` should be changed to `UNKNOWN`. It cannot be null, as spec marks this field as requried.

**Comment by pawel-big-lebowski** [comment]
> Regarding `processingType=BATCH`, Spark application can have both batch and streaming jobs at the same time, so this is arguable. This field cannot be null, as spec marks this field as required.

Yes, having `processingType` non empty is quite a problem here. Not sure how to tackle this. How about adding `NONE` processing type?

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java [hunk 1]
```diff
@@ -193,6 +193,24 @@ void sparkEmitsEnvironmentPropertiesFacet() {
...
+    Dataset<Row> df = createTempDataset();
+
+    Dataset<Row> agg = df.groupBy("a").count();
+    agg.write().mode("overwrite").csv("/tmp/test_data/test_output/");
+    spark.stop();
+
+    List<OpenLineage.RunEvent> events = getEventsEmitted(mockServer);
+
+    // Both Spark application and Spark job events have jobType facet
+    assertThat(events)
```
Comment (pawel-big-lebowski) [inline integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java:207]: Would it make sense to include this assertion in the test below `sparkEmitsApplicationDetailsFacet`?

Comment (dolfinus) [reply integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java:207]: Do you mean move this assertion to `sparkEmitsApplicationDetailsFacet`?

Comment (pawel-big-lebowski) [reply integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java:207]: yes, move current assertion to `sparkEmitsApplicationDetailsFacet` and remove current test.

Comment (dolfinus) [reply integration/spark/app/src/test/java/io/openlineage/spark/agent/SparkGenericIntegrationTest.java:164]: Ok, merged most of these tests into one test with multiple asserts

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**